docs/src/proposals/interest-bearing-tokens.md

```diff
@@ -0,0 +1,377 @@
+wallets, applications, and end-users.
+
+## Proposed Solution
+
+Following the approach taken by Aave, we can lean on a crucial feature of i-tokens:
+although the holding's token amount is constantly changing, each token holder's
+proportion of total supply stays the same. Therefore, we can satisfy the
+fungibility requirement for tokens on proportional ownership. The token
+amount used for token transfers / mints / burns is just an interface, and
+everything is converted to proportional shares inside the program.
+
+Let's go through an example. Imagine we have an i-token mint with a total of
+10,000 proportional shares, a 20% annual interest rate, and 1 share is currently
+worth 100 tokens. If I transfer 1,000 tokens today, the program converts that and moves
+10 shares. One year later, 1 share is worth 120 tokens, so if I transfer
+1,000 tokens, the program converts and only moves 8.33 shares. The same concept
+applies to mints and burns.
+
+Following this model, we never have to worry about compounding interest. We
+simply need to call the right program to perform instructions.
+
+## Required Changes
+
+A new interest-bearing token program is only one component of the overall solution,
+since i-tokens need to be easily integrated in on-chain programs, web apps,
+and wallets.
+
+### Token Program Conformance
+
+Since our solution entails the creation of a new token program, we're opening the
+floodgates for other app developers to do the same. To ensure safe and easy integration
+of new token programs, we need a set of conformance tests based on the current
+SPL token test suite.
+
+A new token program must include all of the same instructions as SPL token,
+but the "unchecked" variants should return errors. Concretely, this means the
+program must implement:
+
+* `InitializeMint`
+* `InitializeHolding` (previously `InitializeAccount`)
+* `Revoke`
+* `SetAuthority`
+* `CloseHolding` (previously `CloseAccount`)
+* `FreezeHolding` (previously `FreezeAccount`)
+* `ThawHolding` (previously `ThawAccount`)
+* `TransferChecked`
+* `ApproveChecked`
+* `MintToChecked`
+* `BurnChecked`
+* `InitializeHoldingWithOwnerAddress` (previously `InitializeAccount2`)
+
+And the program should throw errors for:
+
+* `Transfer`
+* `Approve`
+* `MintTo`
+* `Burn`
+* `InitializeMultisig` (use a more general multisig)
+
+New instructions required:
+
+* `CreateHolding` (TODO name): only performs `Allocate` and `Assign` to self,
+useful for creating a holding when you don't know the program that you're
+interacting with. See the Associated Token Account section for how to use this.
+* `TransferAllChecked`: transfers everything from the source holding to the
+destination holding. Useful for closing a interest-bearing token holding since the
+balance is constantly changing.
+
+There are also new read-only instructions, which write data to a provided account
+buffer:
+
+* `SerializeHoldingToSPL`: write holding data in SPL token format
```

Comment:
Hmm, I rather want to have better name. But there is no alternative name come to mind..  ;)

Reply:
`NormalizeHolding`?

Reply:
The SPL part is important, so I propose `NormalizeToSPLHolding`

Reply:
The SPL part is misleading.  `NormalizeToSolanaProgramLibraryHolding` doesn't really make sense.
SPL is not the same as SPL Token

Reply:
`NormalizeToTokenkeg`? jk 😆 
I suppose to be completely explicit, it should be `NormalizeToSPLTokenHolding`, but imo `NormalizeHolding` is sufficient

Reply:
I stand overruled :smile: will do `NormalizeHolding`